jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -315,7 +319,7 @@ ecma_op_object_delete (ecma_object_t *obj_p, /**< the object */
     {
       JERRY_ASSERT (false);
```

Comment:
There are several such asserts in legacy code, perhaps we could fix all of them in a separate patch.